src/main/java/org/springframework/data/neo4j/repository/query/AbstractNeo4jQuery.java

```diff
@@ -98,11 +101,30 @@ boolean isGeoNearQuery() {
 		return GeoPage.class.isAssignableFrom(returnType);
 	}
 
+	boolean isVectorSearchQuery() {
+		var repositoryMethod = this.queryMethod.getMethod();
```

Comment:
Could `QueryMethod.isSearchQuery()` be useful here?

Would it also make sense to check for the presence of the `@VectorSearch` annotation?